pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -113,6 +115,107 @@ type ReconcileNodeNetworkConfigurationPolicy struct {
 	scheme *runtime.Scheme
 }
 
+// It will return the policies with same labels at the one from argument
+func (r *ReconcileNodeNetworkConfigurationPolicy) filterByNodeLabels(node corev1.Node) (nmstatev1alpha1.NodeNetworkConfigurationPolicyList, error) {
+	policyList := nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+	filteredPolicyList := nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+
+	// Retrieve all the policies
+	err := r.client.List(context.TODO(), &client.ListOptions{}, &policyList)
+	if err != nil {
+		return policyList, err
+	}
+
+	// Get the ones that fit this node
+	for _, policy := range policyList.Items {
+		if nodeSelectorMatchesThisNode(r.client, &policy) {
+			filteredPolicyList.Items = append(filteredPolicyList.Items, policy)
+		}
+	}
+
+	return filteredPolicyList, nil
+}
+
+func unmarshalInterfaces(state nmstatev1alpha1.State) (map[string]interface{}, []interface{}, error) {
+	// Unmarshall interfaces state into unstructured golang
+	var unstructuredState map[string]interface{}
+	err := yaml.Unmarshal(state, &unstructuredState)
+	if err != nil {
+		return unstructuredState, []interface{}{}, fmt.Errorf("error unmarshaling state: %v", err)
+	}
+	return unstructuredState, unstructuredState["interfaces"].([]interface{}), nil
+}
+
+func mapByName(interfacesList []interface{}) (map[string]interface{}, error) {
```

Comment:
having "interface" as in network interface and "interface" as golang thing on one line is killing me... Can we specify a new interface networkInterface/iface to make it easier for poor Petr's eyes? Or rename all variables to be ifaceList etc.